### PR TITLE
fix(stock): improve movements registry

### LINKED ISSUE #2072: bug: Stock Movements cannot filter by input/output
The Stock Movements registry does not filter by input/output.  Even though it lists the radio buttons to do so, the "output" selection does not work.  Selecting either "input" or "output" will filter the registry so that only input records are displayed.

### LINKED ISSUE #2073: bug: Stock Movements Registry does not sort "cost" as a number
The Stock Movements Registry does not sort the `cost` column as a number.  This is due to a lack of `type:number` on the grid options.

### PR DESCRIPTION
**Pull request by jniles**
This commit fixes a number of bugs on the movements registry.
 1. The input/output search parameter now actually filters correctly.
 2. The "reason" ui-select actually filters the registry.
 3. The "actions" column does not present a dropdown on header rows.
 4. The date and unit cost columns sort properly.
 5. The grid footer no longer gives untranslated english text in the
 footer.

It also improves the performance of the registry by pre-computing costs
in the same way that it pre-computes the flux mappings.

Closes #2073.  Closes #2072.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through eslint.
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the online review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by DedrickEnc** [comment]
@jniles
This P.R needs a rebase and then after I will do the review.

**Comment by jniles** [comment]
@DedrickEnc, done.

File: client/src/modules/stock/movements/modals/search.modal.html [hunk 1]
```diff
@@ -20,13 +20,13 @@
...
               <label>
-                <input type="radio" name="is_exit" value="0" ng-model="$ctrl.searchQueries.is_exit"> 
+                <input type="radio" name="is_exit" value="0" ng-model="$ctrl.searchQueries.is_exit">
                 <span translate>STOCK.INPUT</span>
               </label>
             </div>
             <div class="radio">
               <label>
-                <input type="radio" name="is_exit" value="0" ng-model="$ctrl.searchQueries.is_exit"> 
+                <input type="radio" name="is_exit" value="1" ng-model="$ctrl.searchQueries.is_exit">
```
Comment (DedrickEnc) [inline client/src/modules/stock/movements/modals/search.modal.html:29]: 👍 Good catch

File: client/src/modules/stock/movements/modals/search.modal.html [hunk 2]
```diff
@@ -59,27 +59,16 @@
...
             <label class="control-label" translate>STOCK.FLUX</label>
             <bh-clear on-clear="$ctrl.clear('flux_id')"></bh-clear>
-            
+
             <ui-select name="inventory" ng-model="$ctrl.searchQueries.flux_id">
               <ui-select-match>
                 <span translate>{{$select.selected.label}}</span>
               </ui-select-match>
-              <ui-select-choices ui-select-focus-patch repeat="flux as flux in $ctrl.fluxes | filter:{ 'label': $select.search }">
+              <ui-select-choices ui-select-focus-patch repeat="flux.id as flux in $ctrl.fluxes | filter:{ 'label': $select.search }">
```
Comment (DedrickEnc) [inline client/src/modules/stock/movements/modals/search.modal.html:67]: 👍

File: client/src/modules/stock/movements/templates/action.cell.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="ui-grid-cell-contents text-action" uib-dropdown dropdown-append-to-body uib-dropdown-toggle>
+<div class="ui-grid-cell-contents text-action" uib-dropdown dropdown-append-to-body uib-dropdown-toggle ng-if="!row.groupHeader">
```
Comment (DedrickEnc) [inline client/src/modules/stock/movements/templates/action.cell.html:1]: As you suggested in the email, this enhancement should be done in every registry with grouping option. So an issue should be better.

**Review by DedrickEnc — APPROVED** [review]
The cleaning in this P.R is helpful, I am okay to pull it on master.
LGTM

**Comment by jniles** [comment]
bors  r+

**Pull request merged**